modules/persistent-storage-csi-vsphere-top-aware.adoc

```diff
@@ -37,6 +27,74 @@ Datacenter: Atlanta
++
+While vSphere allows you to create categories with any arbitrary name, {product-title} strongly recommends use of `openshift-region` and `openshift-zone` names for defining topology.
++
+For more information about vSphere categories and tags, see the VMware vSphere documentation.
+
+. In {product-title}, create failure domains:
+* xref:../../installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.adoc#installation-installer-provisioned-vsphere-config-yaml_installing-vsphere-installer-provisioned-customizations[During installation]
+
+* xref:../../post_installation_configuration/post-install-vsphere-zones-regions-configuration.adoc#specifying-regions-zones-infrastructure-vsphere_post-install-vsphere-zones-regions-configuration[Post-installation]
+
+. Create a tag to assign to datastores across failure domains:
++
+When an {product-title} spans more than one failure domain, the datastore might not be shared across those failure domains, which is where topology-aware provisioning of persistent volumes (PVs) is useful. 
++
+.. In vCenter, create a category for tagging the datastores. For example, `openshift-zonal-datastore-cat`. You can use any other category name, provided the category uniquely is used for tagging datastores participating in {product-title} cluster. Also, ensure that `StoragePod`, `Datastore`, and `Folder` are selected as Associable Entities for the created category. 
+.. In vCenter, create a tag that uses the previously created category. This example uses the tag name `openshift-zonal-datastore`.
+.. Assign the previously created tag (in this example `openshift-zonal-datastore`) to each datastore in a failure domain that would be considered for dynamic provisioning.
++
+[NOTE]
+====
+You can use any names you like for categories and tags. The names used in this example are provided as recommendations. Ensure that the tags and categories that you define uniquely identify only datastores that are shared with all hosts in the {product-title} cluster.
+====
+
+.Results
+Creating persistent volume claims (PVCs) and PVs from the topology aware storage class are truly zonal, and should use the datastore in their respective zone depending on how pods are scheduled:
+
+[source, terminal]
+----
+~ $ oc get pv <pv-name> -o yaml
+----
+
+.Example output
+
+[source, terminal]
+----
+...
+nodeAffinity:
+  required:
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: topology.csi.vmware.com/openshift-zone <1>
+        operator: In  
+        values:
+        - <openshift-zone>
+      -key: topology.csi.vmware.com/openshift-region <1>
+        operator: In
+        values:
+        - <openshift-region>
+...
+peristentVolumeclaimPolicy: Delete
+storageClassName: <zoned-storage-class-name> <2>
+volumeMode: Filesystem
+...
+----
+<1> PV has zoned keys.
+<2> PV is using the zoned storage class.
+
+== Creating vSphere toplogy without an infra topology
```

Comment:
@gnufied I understand the prerequisite is we don't have more than 2 failure domain in infrastructures, otherwise we are not allowed to have topology both in "failure domain" and "ClusterCSIDriver". See https://issues.redhat.com/browse/STOR-1123.

Reply:
It is not about number of failure-domains though but may be it is worth calling out that:

> "Openshift recommends using infrastructure object for specifying failure-domains in a topology aware setup.  Specifying failure-domains in infrastructure object and specify topology-categories in `ClusterCSIDriver` object at the same time is unsupported operation"

Reply:
Done.